README.md

```diff
@@ -111,31 +116,37 @@ block in the same round a Polka, and a collection of +⅔ PreCommits for a singl
 skipped.  Thus, validators wait some amount TimeoutPropose before they Prevote
-Nil.  Progression through the rest of the round is fully asychronous, in that
+Nil, and the value of TimeoutPropose increases with each round.  
+Progression through the rest of a round is fully asychronous, in that
 progress is only made once a validator hears from +⅔ of the network.
+In practice, it would take an extremely strong adversary to indefinetely thwart the weak synchrony assumption 
+(causing the consensus to fail to ever commit a block), and doing so can be made even more difficult by 
+using randomized values of TimeoutPropose on each validator.
 
 An additional set of constraints, or Locking Rules, ensure that the network will
 eventually commit just one value. Any malicious attempt to cause more than one
 value to be committed can be identified.  First, a PreCommit for a block must
 come with justification, in the form of a Polka for that block. If the validator
 has already PreCommit a block at round `R_1`, we say they are "locked" on that
-block, and the Polka used to justify the new PreCommit at round `R_2` must come
-in a round `R_polka` where `R_1 < R_polka <= R_2`.  Second, validators must
-Propose and/or PreVote the block they are "locked" on.  Together, these
-conditions ensure that a validator does not PreCommit without sufficient
-evidence, and that validators which have already PreCommit cannot contribute to
-evidence to PreCommit something else.  This ensures both safety and liveness of
-the consensus algorithm.
+block, and the Polka used to justify the new PreCommit at round R_2 must come in
+a round `R_polka` where `R_1 < R_polka <= R_2`.  Second, validators must Propose
+and/or PreVote the block they are locked on.  Together, these conditions
+ensure that a validator does not PreCommit without sufficient evidence, and that
+validators which have already PreCommit cannot contribute to evidence to
+PreCommit something else.  This ensures both safety and liveness of the
+consensus algorithm.
+
+The full details of the protocol are described [here](https://github.com/tendermint/tendermint/https://github.com/tendermint/tendermint/wiki/Byzantine-Consensus-Algorithm).
 
 Tendermint’s security derives simultaneously from its use of optimal Byzantine
-fault-tolerance and the locking mechanism.  The former ensures that ⅓ or more
-validators must be Byzantine to cause a violation of safety, where more than two
-values are committed.  The latter ensures that, if ever any set of validators
-attempts, or even succeeds, in violating safety, they can be identified by the
-protocol.  This includes both voting for conflicting blocks and broadcasting
+fault-tolerance and the locking mechanism.  Together, they ensure that:
+* ⅓+ validators must be Byzantine to cause a violation of safety, where more than two
+values are committed.  The latter ensures that, 
```

Comment:
I don't think we should refer to BFT and the locking mechanism as separate things.  Together they are a BFT, and together they ensure...